docs/contributor/image_mode.md

```diff
@@ -0,0 +1,308 @@
+# Image Mode for MicroShift
+
+Image mode is a new approach to operating system deployment that lets users build,
+deploy, and manage Red Hat Enterprise Linux as a bootable container (`bootc`)
+image. Such an image uses standard OCI/Docker containers as a transport and delivery
+format for base operating system updates. A `bootc` image includes a Linux kernel,
+which is used to boot.
+
+MicroShift build and deployment procedures can utilize bootable containers to
+benefit from this technology.
+
+This document demonstrates how to:
+* Create a bootable container image and boot RHEL from this image using `podman`
+* Store a `bootc` image in a remote registry and use it for installing a new RHEL
+operating system
+
+> See [Image mode for Red Hat Enterprise Linux](https://developers.redhat.com/products/rhel-image-mode/overview)
+for more information.
+
+The procedures described below require the following setup:
+* A `RHEL 9.4 host` with an active Red Hat subscription for building MicroShift `bootc`
+images and running containers
+* A `physical hypervisor host` with the [libvirt](https://libvirt.org/) virtualization
+platform for starting virtual machines that run RHEL OS containing MicroShift binaries
+* A `remote registry` (i.e. `quay.io`) for storing and accessing `bootc` images
+
+## Build MicroShift Bootc Image
+
+Log into the `RHEL 9.4 host` using the user credentials that have SUDO
+permissions configured.
+
+### Build Image
+
+Create the `Containerfile` file with the following contents.
+
+```
+FROM registry.redhat.io/rhel9/rhel-bootc:9.4
+
+ARG USHIFT_VER=4.16
+RUN dnf config-manager \
+        --set-enabled rhocp-${USHIFT_VER}-for-rhel-9-$(uname -m)-rpms \
+        --set-enabled fast-datapath-for-rhel-9-$(uname -m)-rpms
+RUN dnf install -y firewalld microshift && \
+    systemctl enable microshift && \
+    dnf clean all
+
+# Create a default 'redhat' user, allowing it to run sudo commands
+ARG USER_PASSWD=\$5\$XDVQ6DxT8S5YWLV7\$8f2om5JfjK56v9ofUkUAwZXTxJl3Sqnc9yPnza4xoJ0
```

Comment:
The chance that someone copy-pastes this is quite high; I don't think it's a good idea to have default passwords. You can make it a required parameter, or just refer to the docs for users.